src/map/clif.c

```diff
@@ -25955,6 +25956,111 @@ static void clif_parse_adventuterAgencyJoinResult(int fd, struct map_session_dat
 #endif
 }
 
+/**
+ * Creates a "item link" tag (<ITEML>) string representing item "it" into buf.
+ * The resulting format and feature support is client-specific (newer clients supports more).
+ * 
+ * @param buf uninitialized buffer where the string will be written to. Caller MUST free it.
+ * @param it item data to be formatted
+ */
+static void clif_format_itemlink(StringBuf *buf, const struct item *it)
+{
+	nullpo_retv(buf);
+	nullpo_retv(it);
+
+	StrBuf->Init(buf);
```

Comment:
I wonder, maybe it would be better to ask the caller to give an initialized buffer? this way the caller may choose to give an already filled buffer and this function would only add extra things to it.

And it would be clear for the caller that it needs to Init + Destroy and not only Destroy

Reply:
I think so. If the caller is passing it, then it'd make sense if it'd be already initialized (that'd also allow a caller to pass a StringBuf that already contains a string it wants the item link appended to).

Otherwise, if you want to initialize it in this function, then I think the signature should be `StringBuf *clif_format_itemlink(const struct item *it)`, so that it's allocated and initialized here (and it's care of the caller to destroy it afterwards), but that'd lose the convenience of appending to an already existing string.

I'm fine with either of these two options

Reply:
thanks! I changed it so the caller needs to send the initialized buffer